xchange-stream-coinmate/src/main/java/info/bitrich/xchangestream/coinmate/CoinmateStreamingService.java

```diff
@@ -0,0 +1,53 @@
+package info.bitrich.xchangestream.coinmate;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import info.bitrich.xchangestream.coinmate.dto.auth.AuthParams;
+import info.bitrich.xchangestream.service.netty.JsonNettyStreamingService;
+import io.reactivex.Observable;
+import io.reactivex.subjects.PublishSubject;
+import sun.rmi.runtime.Log;
+
+import java.io.IOException;
+
+/**
+ * The new API uses one WebSocket connection per channel, unlike other exchanges,
+ * which use channels over one WebSocket. This is a hack to use JsonNettyStreamingService as
+ * single channel source.
+ */
+public class CoinmateStreamingService extends JsonNettyStreamingService {
+
+  private PublishSubject<JsonNode> messages;
+
+  CoinmateStreamingService(String url) {
+    super(url);
+    this.messages = PublishSubject.create();
+  }
+
+  @Override
+  protected String getChannelNameFromMessage(JsonNode message) throws IOException {
+    return null;
+  }
+
+  @Override
+  public String getSubscribeMessage(String channelName, Object... args) throws IOException {
+    return null;
```

Comment:
Better to `throw new UnsupportedOperationException()`

Reply:
An exception thrown here would be sent down the Observable chain in subscribeChannel

Reply:
It would get caught and logged, but returning null from `getSubscribeMessage` eventually gets handled properly in `sendMessage`, binance follows the same pattern.

Reply:
https://github.com/knowm/XChange/blob/develop/xchange-stream-service-netty/src/main/java/info/bitrich/xchangestream/service/netty/NettyStreamingService.java#L378